src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -62,13 +63,32 @@
     private final String apiUri;
     private String pattern = ".*";
 
+    private @CheckForNull String includes;
+    private @CheckForNull String excludes;
+
     @DataBoundConstructor public GitHubSCMNavigator(String apiUri, String repoOwner, String scanCredentialsId, String checkoutCredentialsId) {
         this.repoOwner = repoOwner;
         this.scanCredentialsId = Util.fixEmpty(scanCredentialsId);
         this.checkoutCredentialsId = checkoutCredentialsId;
         this.apiUri = Util.fixEmpty(apiUri);
     }
 
+    public @Nonnull String getIncludes() {
+        return includes != null ? includes : DescriptorImpl.defaultIncludes;
+    }
+
+    @DataBoundSetter public void setIncludes(@Nonnull String includes) {
+        this.includes = includes.equals(DescriptorImpl.defaultIncludes) ? null : includes;
+    }
+
+    public @Nonnull String getExcludes() {
```

Comment:
Please do not use the bug marker for purely stylistic issues.

(In this case the underlying question is whether `@Nonnull` could become a type qualifier annotation in the foreseeable future. These would be placed immediately before the type, here the method return type, since they are annotating the type not the method as a whole. In the case of parameters, as in `setIncludes` above, the difference between an annotation on the parameter as a whole vs. an annotation on the parameter’s type is still real, but not visible in the syntax of the program, unless the parameter happens to have some qualifiers like `final`.)